src/meta/src/barrier/progress.rs

```diff
@@ -192,7 +254,10 @@ impl CreateMviewProgressTracker {
         // If the target command found in progress map, return and remove it. Note that the command
         // should have finished if not found.
         if let Some(Some(epoch)) = epochs.first() {
-            Some(self.progress_map.remove(epoch).unwrap().1)
+            match self.progress_map.remove(epoch).unwrap().1 {
+                StreamJobOrigin::TrackingCommand(t) => Some(t),
+                _ => None,
```

Comment:
Have you tried to cancel a recovered streaming job locally? I'm not sure if it works or not if returns a `None` here. The command will be used to clear some status in checkpoint controller.

Reply:
Yes, there's a test for it: https://github.com/risingwavelabs/risingwave/pull/12167/files/cff4eabe913a55cbfb8652e08d404c69f178122c#r1361701967

Reply:
Oh I see, checkpoint controller don't contain any status for recovered jobs. The cancel should works and the spawn task in `recover_mview_progress` will get an `channel closed` error and ignore it.